### PR TITLE
Add `schedmd-slurm-gcp-v6-nodeset-dynamic` module

### PR DESCRIPTION
**Pull request by mr0re1**
#### Motivation:

To simplify usage of "dynamic" instances

#### Usage:

```yaml
  - id: dynamic_ns
    source: community/modules/compute/schedmd-slurm-gcp-v6-nodeset-dynamic
    use: [network, controller]
    settings:
      machine_type: n2-standard-2

  - id: dynamic_partition
    source: community/modules/compute/schedmd-slurm-gcp-v6-partition
    use: [dynamic_ns]
    settings:
      partition_name: mp
      is_default: true

  - id: controller
    source: community/modules/scheduler/schedmd-slurm-gcp-v6-controller
    use: [network, dynamic_partition]  

  - id: mig
    source: community/modules/compute/mig
    settings:
      versions:
      - name: highlander # there can be only one
        instance_template: $(dynamic_ns.instance_template_self_link)
      base_instance_name: $(dynamic_ns.node_name_prefix)```

### CODE REVIEW

**Comment by nick-stroud** [comment]
I am approving as I do not want to hold this up while I am out. If you strongly disagree with comments and want to move forward please get second opinion as tiebreaker.

**Review by nick-stroud — APPROVED** [review]

File: community/modules/compute/schedmd-slurm-gcp-v6-dynamic-node-template/README.md
Comment (mr0re1) [reply community/modules/compute/schedmd-slurm-gcp-v6-dynamic-node-template/README.md:39]: > Would it be possible to allow this configuration via use? 

Definetly, but I intentionally didn't name inputs/outputs the same, to keep MIG module - "pure MIG as is" and template module - "I don't know what is MIG".

**Pull request merged**
